drivers/serial/uart_stm32.c

```diff
@@ -675,10 +677,27 @@ static int uart_stm32_init(struct device *dev)
 	LL_USART_SetTransferDirection(UartInstance,
 				      LL_USART_DIRECTION_TX_RX);
 
-	/* 8 data bit, 1 start bit, 1 stop bit, no parity */
+	/* Determine the parity, if we use other than 'none' we must use 9B
+	 *  data (to get 8 data + 1 parity)
+	 */
+	if (config->parity == 2) {
+		ll_parity = LL_USART_PARITY_EVEN;
+		ll_datawidth = LL_USART_DATAWIDTH_9B;
+	} else if (config->parity == 1) {
+		ll_parity = LL_USART_PARITY_ODD;
+		ll_datawidth = LL_USART_DATAWIDTH_9B;
+	} else {  /* Default to 8N0, but show warning if invalid value */
+		if (config->parity != 0) {
+			LOG_ERR("Invalid parity setting '%d'", config->parity);
```

Comment:
LOG_WRN to show this warning instead of LOG_ERR ?

Reply:
Check, changed.